feat(trace-eap-waterfall) Loading eap trace-item details in useTraceR…

Abdkhan14 opened the pull request:
This PR updates `useTraceRootEvent` to be able to fetch both transaction event details and trace item details (i.e. eap spans and logs).

Bulk of the changes are from accepting the new return type from `useTraceRootEvent`.
Also, instead of fetching the details for the first event in the trace we now use the event that best represents the trace. 

Sideeffects: 
- We now load event higlights from attributes (just environment and release for now)
- We now load tags tree from eap span attributes

<img width="1184" alt="Screenshot 2025-04-30 at 3 33 38 PM" src="https://github.com/user-attachments/assets/8ef6dad2-ecab-43f6-821c-a8a9d0189cb3" />

Commit:
feat(trace-eap-waterfall) Loading eap trace-item details in useTraceRootEvent

Commit:
feat(trace-eap-waterfall): Fixing tests

Abdkhan14 committed:
Merge branch 'master' into abdk/eap-trace-root-event

Commit:
Merge branch 'master' into abdk/eap-trace-root-event

Commit:
feat(trace-eap-waterfall): Enabling eap query for logs only views

Commit:
feat(trace-eap-waterfall): Fixing tests

edwardgou-sentry commented on static/app/views/performance/newTraceDetails/traceApi/utils.tsx:
```diff
@@ -8,3 +16,81 @@ export function isEmptyTrace(trace: TraceTree.Trace): boolean {
+      )
+    ) {
+      candidateEvent = event;
+      continue;
```
should we just use `break` instead? if we're returning `event: firstRootEvent ?? candidateEvent ?? firstEvent,` then we will never use `firstEvent` anyways if we have a `candidateEvent`.

Abdkhan14 replied:
@edwardgou-sentry but we might find a rootEvent after we find a candidate

edwardgou-sentry commented on static/app/views/performance/newTraceDetails/traceApi/utils.tsx:
```diff
@@ -8,3 +16,81 @@ export function isEmptyTrace(trace: TraceTree.Trace): boolean {
+  const events = isTraceNode(traceNode)
+    ? [...traceNode.value.transactions, ...traceNode.value.orphan_errors]
+    : traceNode.value;
+  for (const event of events) {
```
nit: this for loops feels a little hard to read. You might have an easier time using something like to the following:
```
const firstRootEvent = events.find(isRootEvent);
```
And something similar for `candidateEvent` and `firstEvent`.

edwardgou-sentry commented on static/app/views/performance/newTraceDetails/traceApi/useTraceRootEvent.tsx:
```diff
@@ -1,26 +1,60 @@
+  const enabledBase =
+    !treeIsLoading &&
+    (tree.type === 'trace' || hasOnlyLogs) &&
+    !!rep &&
+    !!rep.event &&
```
typescript says `rep` is never undefined, so you can probably skip checking for `!!rep`. `!!rep?event` should also work if you prefer that

Abdkhan14 replied:
Valid point, I'll push the change

Abdkhan14 commented on static/app/views/performance/newTraceDetails/traceHeader/projects.tsx:
```diff
@@ -32,13 +31,19 @@ function Projects({projects, logs, tree}: Props) {
-        new Set(logs.map(log => projectIdToSlug.get(log[OurLogKnownFieldKey.PROJECT_ID])))
-      ).filter(defined);
+      for (const log of logs) {
+        const projectId = String(log[OurLogKnownFieldKey.PROJECT_ID]);
```
Yep I like the functional implementation better, I'll push.

scttcper commented on static/app/views/performance/newTraceDetails/traceActionsMenu.tsx:
```diff
@@ -38,9 +39,20 @@ function TraceActionsMenu({
-  const traceProject = rootEventResults.data
-    ? projects.find(p => p.id === rootEventResults.data.projectID)
-    : undefined;
+  let traceProject;
```
types

Commit:
feat(trace-eap-waterfall): Addressing PR suggestions

Abdkhan14 committed:
feat(trace-eap-waterfall): Building trace highlights from eap trace-item attrs (#90755)

<img width="1253" alt="Screenshot 2025-05-01 at 3 01 05 PM"
src="https://github.com/user-attachments/assets/f11ab746-553a-401e-b14a-e7a263c34af6"
/>

---------

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

Commit:
feat(trace-eap-waterfall): Minor cleanup
